src/HuC3.md

```diff
@@ -0,0 +1,171 @@
+# HuC-3
+
+HuC-3 is an MBC used by some Game Boy games which, besides doing the
+usual MBC stuff, also provides a real-time clock, speaker, and infrared
+communication. The CR2025 coin cell is user-replaceable. It is a
+successor to the [HuC1](./HuC1.md).
+
+The HuC-3 is poorly understood. Observed behavior suggests the
+real-time clock and tone generator are implemented using a 4-bit
+microcontroller core with internal program ROM.
+
+
+## Memory
+
+### 0000–3FFF – ROM Bank 00 (Read-Only)
+
+Contains the first 16 KiB of the ROM.
+
+### 4000–7FFF – ROM Bank 00–7F (Read-Only)
+
+This area may contain any of the further 16 KiB banks of the ROM. It is
+unknown whether bank $00 can be selected here.
+
+### A000–BFFF – RAM Bank 00–03, or RTC/IR register (Read/Write)
+
+Depending on the current register selection and RAM Bank Number (see
+below), this memory space is used to access an 8 KiB external RAM Bank,
+or a single I/O Register.
+
+
+## Memory Control Registers
+
+### 0000–1FFF – RAM/RTC/IR Select (Write-Only)
+
+Writing to this register maps cart RAM, RTC registers or IR registers
+into memory at A000–BFFF. Only the lower 4 bits are significant.
+
+Value | Feature
+------|-----------------------------
+  $0  | Cart RAM (read-only)
+  $A  | Cart RAM (read/write)
+  $B  | RTC command/argument (write)
+  $C  | RTC command/response (read)
+  $D  | RTC semaphore (read/write)
+  $E  | IR (read/write)
+
+If any other value is written, reads from A000–BFFF return open bus
+values (often $FF, but not guaranteed), and writes are ignored.
+
+### 2000–3FFF – ROM Bank Number (Write-Only)
+
+HuC-3 can accept a 7-bit bank number here.
+
+### 4000–5FFF – RAM Bank Select (Write-Only)
+
+HuC-3 can accept a bank number of at least 2 bits here.
+
+### 6000–7FFF – Nothing? (Write-Only)
+
+Games write $01 here on startup, but it doesn\'t seem to have any
+observable effect.
+
+
```

Comment:
```suggestion
```

Reply:
I’ve intentionally left two blank lines before every level 2 heading (`##` prefix) besides the first to make the sections visible when editing the source in a text editor.

Reply:
AFAIK, that is not valid Markdown (and I have seen parsers act incorrectly in that case), so I'd rather avoid it.

Reply:
You know, I really hate markdown because every single implementation of it is different – it’s the opposite of a standard.  I’ll remove them here.

Reply:
Well, that's what [CommonMark](https://commonmark.org) attempts to fix, but bugs are legion.

Reply:
> Well, that's what [CommonMark](https://commonmark.org/) attempts to fix, but bugs are legion.

I’m not even sure it’s “bugs” as such – the different dialects are evolving in different directions for whatever reason.

For example GitHub markdown doesn’t require a blank line to introduce a list, treats unadorned newlines as hard line breaks, and so in.  It seems to be an intentional break with the other flavours.

RST is also evolving in its own direction, with its own syntax for links, anchors, invoking macros, etc.  It seems to be more influenced by the TeX/LaTeX family of document markup languages.  We end up needing to lean pretty heavily on some of its features for the MAME documentation.

Anyway, that’s all kind of beside the point.  All the differences just make me nervous about writing anything in markdown when it’s going to be rendered by an unfamiliar engine.